### PR TITLE
[apollo-upload-client] Update dependencies

### PR DESCRIPTION
**Pull request by martin-badin**
Discussion: #54974

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test apollo-upload-client`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by martin-badin** [comment]
@peterblazejewicz please review

File: types/apollo-upload-client/package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
     "private": true,
     "dependencies": {
-        "@apollo/client": "^3.1.3",
-        "graphql": "^15.3.0"
+        "@apollo/client": "^3.5.6",
+        "graphql": "^15.8.0"
```
Comment (peterblazejewicz) [inline types/apollo-upload-client/package.json:5]: should these ones be same as original versions?
```ts
"@apollo/client": "^3.0.0",
"graphql": "14 - 16"
```
https://github.com/jaydenseric/apollo-upload-client/blob/master/package.json

**Comment by peterblazejewicz** [comment]
those files are using CJS exports, only by ESM convention supporting `default`, so I think types should reflect source code, e.g.:
```ts
declare function createUploadLink(...)

// I know it's cumbersome
declare namespace createUploadLink {
    type UploadLinkOptions ....
}

exports = createUploadLink;
```
https://unpkg.com/browse/apollo-upload-client@17.0.0/public/createUploadLink.js

File: types/apollo-upload-client/tsconfig.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
         "lib": [
             "es6",
             "dom",
-            "esnext.asynciterable"
+            "esnext"
```
Comment (peterblazejewicz) [inline types/apollo-upload-client/tsconfig.json:7]:
```suggestion
            "ES2018.AsyncGenerator",
            "ES2018.AsyncIterable"
```
it looks it needs only those

**Review by peterblazejewicz — APPROVED** [review]
LGTM!

**Comment by martin-badin** [comment]
Ready to merge

**Pull request merged**